langchain4j-chatglm/src/main/java/dev/langchain4j/model/chatglm/ChatGLMChatModel.java

```diff
@@ -0,0 +1,71 @@
+package dev.langchain4j.model.chatglm;
+
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.output.Response;
+import lombok.Builder;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static dev.langchain4j.internal.RetryUtils.withRetry;
+import static dev.langchain4j.internal.Utils.getOrDefault;
+
+public class ChatGLMChatModel implements ChatLanguageModel {
+
+    private final ChatGLMClient client;
+    private final Double temperature;
+    private final Double topP;
+    private final Integer maxLength;
+    private final Integer maxRetries;
+
+    @Builder
+    public ChatGLMChatModel(String baseUrl, Duration timeout,
+                            Double temperature, Integer maxRetries,
+                            Double topP, Integer maxLength) {
+        this.client = new ChatGLMClient(baseUrl, timeout);
+        this.temperature = getOrDefault(temperature, 0.7);
+        this.maxRetries = getOrDefault(maxRetries, 3);
+        this.topP = topP;
+        this.maxLength = maxLength;
+    }
+
+
+    @Override
+    public Response<AiMessage> generate(List<ChatMessage> messages) {
```

Comment:
Please validate explicitly that no `SystemMessage` is present in `messages` (if it is not supported by ChatGLM)

Reply:
done